### PR TITLE
feat(Image): allow to add images to the style

### PR DESCRIPTION
**Pull request by device25**
I added the Image component. Now you can add, replace and delete images in the map style. Can you review this feature?

### CODE REVIEW

File: src/components/Image/index.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  /** The ID of the image. */
+  id: string,
+
+  /**
+   * The image as an `HTMLImageElement`, `ImageData`, object with `width`,
+   * `height`, and `data`
+   * properties with the same format as `ImageData` or string with image url
+   * */
+  image:
+    | window.HTMLImageElement
```
Comment (stepankuzmin) [inline src/components/Image/index.js]: One should not use `window` in type definitions

File: src/components/Image/index.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    | { width: number, height: number, data: Uint8Array | Uint8ClampedArray }
+    | StyleImageInterface
+    | string,
+
+  /**
+   * options.pixelRatio The ratio of pixels in the image to physical pixels on
+   * the screen
+   * options.sdf Whether the image should be interpreted as an SDF image
+   */
+  options?: {
```
Comment (stepankuzmin) [inline src/components/Image/index.js]: Please, flatten options into separate props.

File: src/components/Image/index.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    if (typeof image === 'string') {
+      map.loadImage(image, (error, img) => {
+        if (error) {
+          throw error;
+        }
+
+        map.addImage(id, img, options);
+      });
+    } else {
```
Comment (stepankuzmin) [inline src/components/Image/index.js]: Use the "return early" here

File: src/components/Image/index.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      map.addImage(id, image, options);
+    }
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const map = this._map;
+    const { image } = this.props;
+
+    if (image !== prevProps.image) {
```
Comment (stepankuzmin) [inline src/components/Image/index.js]: Remove negotiation and return early here

File: src/components/Image/index.js (hunk 1 above)
Comment (device25) [reply src/components/Image/index.js]: I can use mapbox types like they do it [here](https://github.com/mapbox/mapbox-gl-js/blob/master/src/ui/map.js#L6). Is it ok?

File: src/components/Image/index.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    super(props);
+    this._id = props.id;
+  }
+
+  componentDidMount() {
+    const map = this._map;
+    const { id, image, pixelRatio, sdf } = this.props;
+    const options = { pixelRatio, sdf };
+
+    if (typeof image === 'string') {
```
Comment (stepankuzmin) [inline src/components/Image/index.js]: type constraint duplicated 3 times, consider refactoring

**Pull request merged**